content/blog/2020-06-29-june-20-community-gems.md

```diff
@@ -0,0 +1,164 @@
+_Just like this but with technical documentation._
+
+### Q: After I pushed my local data to remote storage, I noticed the file names are different in my storage repository- they're hash values. [Can I make them more meaningful names?](https://discord.com/channels/485586884165107732/563406153334128681/717737163122540585)
+
+No, but for a good reason! What you're seeing are cached files, and they're
+stored with a special naming convention that makes DVC versioning and addressing
+possible- these file names are how DVC deduplicates data (to avoid keeping
+multiple copies of the same file version) and ensures that each unique version
+of a file is immutable. If you manually overwrote those filenames you would risk
+breaking Git version control. You can
+[read more about how DVC uses this file format in our docs](https://dvc.org/doc/user-guide/dvc-files-and-directories#structure-of-cache-directory).
+
+It sounds like you're looking for ways to interact with DVC-tracked objects at a
+high level of abstraction, meaning that you want to interface with the original
+filenames and not the machine-generated hashes used by DVC. There are a few
+secure and recommended ways to do this:
+
+- If you want to see a human-readable list of files that are currently tracked
+  by DVC, try the `dvc list`
+  command-[read up on it here](https://dvc.org/doc/command-reference/list).
+- Check out our
+  [data registry tutorial](https://dvc.org/doc/use-cases/data-registries#data-registries)
+  to see how the commands `dvc get` and `dvc import` are used to download and
+  share DVC-tracked artifacts. The syntax is built for an experience like using
+  a package manager.
+- The [DVC Python API](https://dvc.org/doc/api-reference) gives you programmatic
+  access to DVC-tracked artifacts, using human-readable filenames.
+
+### Q: [Is it better practice to `dvc add` data files individually, or to add a directory containing multiple data files?](https://discord.com/channels/485586884165107732/563406153334128681/722141190312689675)
+
+If the directory you're adding is logically one unit (for example, it is the
+whole dataset in your project), we recommend using `dvc add` at the directory
+level. Otherwise, add files one-by-one. You can
+[read more about how DVC versions directories in our docs](https://dvc.org/doc/user-guide/dvc-files-and-directories#structure-of-cache-directory).
+
+### Q: [Do you have any examples of using DVC with MinIO?](https://discord.com/channels/485586884165107732/563406153334128681/722780202844815362)
+
+We don't have any tutorials for this use case exactly, but it's a very
+straightforward modification from
+[our basic use cases](https://dvc.org/doc/use-cases). The key difference when
+using MinIO or a similar S3-compatible storage (like DigitalOcean Spaces or IBM
+Cloud Object Storage) is that in addition to setting remote data storage, you
+must set the `endpointurl` too. For example:
+
+```dvc
+$ dvc remote add -d myremote s3://mybucket/path/to/dir
```

Comment:
isn't it best to use long flag names for commands in documentation? `--default` better than `-d`? Otherwise people may accidentally change their default remote when copy-pasting this command.